### PR TITLE
feat: update default inflation rate

### PR DESCRIPTION
**Pull request by matthiasmatt**
# Purpose / Abstract

- Update inflation params to inflate 800M NIBI over 8 years.

## Summary by CodeRabbit

- **Bug Fixes**
	- Adjusted expected values in test cases for more accurate inflation calculations.
- **New Features**
	- Updated inflation parameters including polynomial factors and distribution percentages for a refined inflation model.
- **Documentation**
	- Documented the inflation fix in the project's changelog.

### CODE REVIEW

**Comment by Unique-Divine** [comment]
Where are the numbers coming from? I'm seeing different coefficients when running the code that generates this: https://github.com/NibiruChain/tokenomics

**Comment by k-yang** [comment]
> Where are the numbers coming from? I'm seeing different coefficients when running the code that generates this: https://github.com/NibiruChain/tokenomics

It's because the `tokenomics` script assumes `810.6M` inflation but that number was changed in the latest tokenomics spreadsheet to `807.735M`. 

Referring to [this Slack thread](https://nibiruchain.slack.com/archives/C051XRS2PKR/p1708360356726489?thread_ts=1708100962.467499&cid=C051XRS2PKR)

File: x/inflation/types/inflation_calculation_test.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ var ExpectedYearlyInflation = []sdk.Dec{
 // = (60% of the total supply) - (Community supply at start)
 // = (60% of 1.5 billion) - (Community supply at start)
 // = 810.6 million NIBI
-var ExpectedTotalInflation = sdk.NewDec(810_600_000_000_000)
+var ExpectedTotalInflation = sdk.NewDec(807_735_002_000_000)
```
Comment (k-yang) [inline x/inflation/types/inflation_calculation_test.go:30]: Shouldn't this be 807.735M flat?

Comment (matthiasmatt) [reply x/inflation/types/inflation_calculation_test.go:30]: there was a rounding issue which led to 2 additional nibi. Updating that!

**Review by k-yang — APPROVED** [review]

**Pull request merged**